ecms_base/themes/custom/ecms/ecms.theme

```diff
@@ -555,3 +555,158 @@ function ecms_preprocess_paragraph__numbered_step_item(&$variables): void {
+      'term_name' => $parent_term->get('name')->value,
+      'child_terms' => [],
+      'persons' => _ecms_query_persons_by_category_tid($parent_tid),
+    ];
+
+    // Load child terms.
+    $child_terms = $term_storage->loadChildren($parent_tid);
+
+    foreach ($child_terms as $child_term) {
+      // Check to see if this child term has person nodes associated.
+      $child_term_persons = _ecms_query_persons_by_category_tid($child_term->get('tid')->value);
+
+      if (!empty($child_term_persons)) {
+        // Store the child term's person teasers on parent list.
+        $term_person_list['child_terms'][$child_term->get('name')->value] = $child_term_persons;
+      }
+    }
+
+    $variables['person_listing'] = $term_person_list;
+    $variables['list_mode'] = "term-list";
+  }
+}
+
+/**
+ * Helper function that returns a list of person teasers for a category term.
+ *
+ * @param string $tid
+ *   The term id to query related persons.
+ *
+ * @return array
+ *   A render array of person teasers.
+ */
+function _ecms_query_persons_by_category_tid($tid): array {
+  if (!$tid) {
+    return [];
+  }
+
+  // Get necessary services.
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $node_storage = $entity_type_manager->getStorage('node');
+  $node_builder = $entity_type_manager->getViewBuilder('node');
+
+  // See if any person nodes are tied directly to term.
+  $query = $node_storage->getQuery();
+  $query->condition('type', 'person')
+    ->condition('field_person_category', $tid)
+    ->condition('status', 1)
+    ->sort('field_person_last_name', "ASC");
+
+  $nids = $query->execute();
+
+  $teasers = [];
+
+  // Load teaser displays for each person.
+  if (!empty($nids)) {
+    $nodes = $node_storage->loadMultiple($nids);
+    $teasers = $node_builder->viewMultiple($nodes, 'teaser', \Drupal::languageManager()->getCurrentLanguage()->getId());
+  }
+
+  return $teasers;
+}
+
+/**
+ * Implements hook_preprocess_node().
+ */
+function ecms_preprocess_node__person(&$variables) {
+  $node = $variables['node'];
+
+  // Get necessary services.
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $term_storage = $entity_type_manager->getStorage('taxonomy_term');
+
+  // Get current language.
+  $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+  // Get each term reference.
+  $categories = $node->get('field_person_category')->referencedEntities();
+  $variables['categories'] = [];
+
+  // Loop over each term reference.
+  foreach ($categories as $category) {
+    $category_tid = $category->id();
+
+    // Load parent terms.
+    $category_parent = $term_storage->loadParents($category_tid);
+
+    if (!empty($category_parent)) {
+      // Get first item of keyed array.
+      $category_parent = reset($category_parent);
+
+      // Get translated version of term.
+      $category_parent_trans = \Drupal::service('entity.repository')->getTranslationFromContext($category_parent, $language);
```

Comment:
Will this return default language if there's no translation?

Reply:
Yeah seems to handle it: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21EntityRepository.php/function/EntityRepository%3A%3AgetTranslationFromContext/8.2.x